server/routes/allocation-dashboard/handlers/endDate.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from 'express'
+import { Expose, plainToInstance, Type } from 'class-transformer'
+import { IsNotEmpty, ValidateNested } from 'class-validator'
+import SimpleDate, { simpleDateFromDate } from '../../../commonValidationTypes/simpleDate'
+import IsValidDate from '../../../validators/isValidDate'
+import { convertToTitleCase, formatDate } from '../../../utils/utils'
+import { AllocationUpdateRequest } from '../../../@types/activitiesAPI/types'
+import ActivitiesService from '../../../services/activitiesService'
+import PrisonService from '../../../services/prisonService'
+import DateIsAfterOtherProperty from '../../../validators/dateIsAfterOtherProperty'
+
+export class EndDate {
+  @Expose()
+  @Type(() => SimpleDate)
+  @ValidateNested()
+  @IsNotEmpty({ message: 'Enter a valid end date' })
+  @IsValidDate({ message: 'Enter a valid end date' })
+  @DateIsAfterOtherProperty('startDate', { message: 'Enter a date after the start date' })
+  endDate: SimpleDate
+
+  @Expose()
+  startDate: string
+}
+
+export default class EndDateRoutes {
+  constructor(private readonly activitiesService: ActivitiesService, private readonly prisonService: PrisonService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { allocationId } = req.params
+    const allocation = await this.activitiesService.getAllocation(+allocationId, user)
+    const { scheduleId } = allocation
+    const { prisonerNumber } = allocation
+    const endDate = allocation.endDate ? simpleDateFromDate(new Date(allocation.endDate)) : undefined
+    const prisoner = await this.prisonService.getInmateByPrisonerNumber(prisonerNumber, user)
+    const prisonerName = convertToTitleCase(`${prisoner.firstName} ${prisoner.lastName}`)
```

Comment:
Just a note, but in nunjucks we have a filter `firstNameLastName` which will pull the firstName and lastName from the prisoner object. So you could just pass the whole prisoner object to the view then let nunjucks handle formatting the name with:

`{{ prisoner | firstNameLastName | toTitleCase }}`

Reply:
Noted